pynucastro/rates/reaclib_rate.py

```diff
@@ -779,11 +787,20 @@ def eval(self, T, *, rho=None, comp=None):
             f = s.f()
             r += f(tf)
 
+        scor = 1.0
+        if screen_func is not None:
+            if rho is None or comp is None:
+                raise ValueError("rho (density) and comp (Composition) needs to be defined when applying electron screening.")
+            scor = self.evaluate_screening(rho, T, comp, screen_func,
+                                           symmetric_screening=symmetric_screening)
+
+        r *= scor
+
         return r
 
     def eval_deriv(self, T, *, rho=None, comp=None):
-        """Evaluate the derivative of reaction rate with respect to T
-
+        """Evaluate the derivative of reaction rate with respect to T.
```

Comment:
this function is never used, so we should consider removing it (in a separate PR)